components/script/dom/htmlformelement.rs

```diff
@@ -559,8 +582,28 @@ impl HTMLFormElement {
                     HTMLElementTypeId::HTMLInputElement => {
                         let input = child.downcast::<HTMLInputElement>().unwrap();
                         // Step 3.2-3.7
-                        if let Some(datum) = input.form_datum(submitter) {
-                            data_set.push(datum);
```

Comment:
Yep, this is a problem. function `form_datum` returns only one datum, but in file case, we need to append several such datums, as spec:

> .... then for each file selected in the input element, append an entry to the form data set with the name as the name ....